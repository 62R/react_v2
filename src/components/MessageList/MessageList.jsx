import { Message } from '../Message/Message';
import style from './MessageList.module.scss';

export const MessageList = ({ messageList }) => {
  let prevAuthor = '';
  return (
    <div className={style.messagelist}>
      {messageList.map((message, index) => {
        const messageEl = (
          <Message
            key={index}
            author={message.author}
            text={message.text}
            isOtherAuthor={message.author !== prevAuthor}
            isFirstMessage={index === 0}
          />
        );
        prevAuthor = message.author;
        return messageEl;
      })}
    </div>
  );
};
