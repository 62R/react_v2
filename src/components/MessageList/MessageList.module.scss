.messagelist {
  display: flex;
  flex-direction: column;
  background-color: gainsboro;
  padding: 10px;
  border-radius: 15px;
  font-size: 14px;
}