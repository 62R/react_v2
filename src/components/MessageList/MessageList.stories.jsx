import React from 'react';
import { MessageList } from './MessageList';

export default {
  title: 'MessaggeList',
  component: MessageList,
};

const Template = (args) => <MessageList {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  messageList: [
    {
      author: 'Bot',
      text: 'Wait a massage',
    },
    {
      author: 'Tom',
      text: 'Hi, Bot',
    },
    {
      author: 'Tom',
      text: 'How are you?',
    },
  ],
};
