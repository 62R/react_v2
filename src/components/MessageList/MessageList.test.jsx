import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MessageList } from './MessageList';

const messageList = [
  {
    author: 'Bot',
    text: 'Test1',
  },
  {
    author: 'Tom',
    text: 'Test2',
  },
  {
    author: 'Tom',
    text: 'Test3',
  },
];

describe('Textarea', () => {
  it('render component', () => {
    render(<MessageList messageList={messageList} />);
    expect(screen.getByText('Bot')).toBeInTheDocument();
    expect(screen.getAllByText('Tom')).toHaveLength(1);
    expect(screen.getByText('Test1')).toBeInTheDocument();
    expect(screen.getByText('Test2')).toBeInTheDocument();
    expect(screen.getByText('Test3')).toBeInTheDocument();
  });
  it('render component with spanshot', () => {
    const { asFragment } = render(<MessageList messageList={messageList} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
