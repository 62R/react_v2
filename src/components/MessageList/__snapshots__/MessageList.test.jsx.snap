// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea render component with spanshot 1`] = `
<DocumentFragment>
  <div
    class="messagelist"
  >
    <div
      class="message message_align_bot message_margin_otherauthor message_order_first"
    >
      <p
        class="author"
      >
         Bot 
      </p>
      <p
        class="text"
      >
         Test1 
      </p>
    </div>
    <div
      class="message message_align_you message_margin_otherauthor"
    >
      <p
        class="author"
      >
         Tom 
      </p>
      <p
        class="text"
      >
         Test2 
      </p>
    </div>
    <div
      class="message message_align_you message_margin_sameauthor"
    >
      <p
        class="text"
      >
         Test3 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
